### PR TITLE
dosdebug: Add display of UMBs to MCBs command

### PR DESCRIPTION
**Pull request by andrewbird**
~~~
dosdebug> mcbs
dosdebug>
ADDR      PARAS  OWNER
0284:0000 0x0020 [DOS]
02a5:0000 END

ADDR(UMA) PARAS  OWNER
9fff:0000 0x2300 [DOS]
c300:0000 0x0130 [DOS]
c431:0000 0x0006 [FREE]
c438:0000 0x12c1 [COMMAND]
d6fa:0000 0x0904 [FREE]
dfff:0000 0x1000 [DOS]
f000:0000 0x0513 [FREE]
f514:0000 0x0619 [COMMAND]
fb2e:0000 0x0090 [COMMAND]
fbbf:0000 END(UMA)
~~~

### CODE REVIEW

File: src/plugin/debugger/mhpdbgc.c [hunk 1]
```diff
@@ -981,6 +983,22 @@ static void mhp_mcbs(int argc, char *argv[])
   } else {
     mhp_printf("MCB chain corrupt - missing final entry\n");
   }
+
+  // UMBs
+  for (first = 1, seg = 0x9fff, mcb = MK_FP32(seg, 0); mcb->id == 'M'; /* */) {
```
Comment (stsp) [inline src/plugin/debugger/mhpdbgc.c]: Can we avoid hardcoding 0x9fff?
It won't work with non-default `$_dosmem`.
FreeDOS puts this address into LoL:66h,
which seems to be a non-standard location.
How about just looking past the first 'Z' mcb?
It should point exactly to this one. We can
check if there is an `M` mcb after the first 'Z'
mcb, which will mean UMB.

**Comment by andrewbird** [comment]
Okay, I'll have a look at that.

File: src/plugin/debugger/mhpdbgc.c (hunk 1 above)
Comment (andrewbird) [reply src/plugin/debugger/mhpdbgc.c]: I updated the commit, now finds it at 0x9fff with default values. With dosmem = 600, FDPP fails to find command.com and output of `mcbs` command is
~~~
dosdebug> mcbs
dosdebug> 
ADDR      PARAS  OWNER
02a3:0000 0x0016 [DOS]
02ba:0000 END

ADDR(UMA) PARAS  OWNER
95ff:0000 0x3700 [DOS]
MCB chain corrupt - missing final entry in UMA
~~~

Is that to be expected?

**Comment by skitt** [comment]
Wouldn’t it be better to use [int 21h function 5803h](http://www.delorie.com/djgpp/doc/rbinter/id/21/30.html) to ask DOS to link the MCB chains?

**Comment by stsp** [comment]
This pr is about debugger.
Debugger should not affect the
running system by doing any calls.

**Comment by stsp** [comment]
Andrew, could you please open
bug reports when finding bugs?

**Comment by skitt** [comment]
> This pr is about debugger.
> Debugger should not affect the
> running system by doing any calls.

Oh, yes, sorry!

**Comment by andrewbird** [comment]
Can do, but wanted to be sure it wasn't an invalid thing to do this. Is it an FDPP or dosemu bug?

**Comment by stsp** [comment]
Thanks!

**Pull request merged**